Sources/OpenAPIRuntime/URICoder/Parsing/URIParser.swift

```diff
@@ -205,6 +207,50 @@ extension URIParser {
             }
         }
     }
+    
+    /// Parses the root node assuming the raw string uses the deepObject style
+    /// and the explode parameter is enabled.
+    /// - Returns: The parsed root node.
+    /// - Throws: An error if parsing fails.
+    private mutating func parseExplodedDeepObjectRoot() throws -> URIParsedNode {
+        try parseGenericRoot { data, appendPair in
+            let keyValueSeparator: Character = "="
+            let pairSeparator: Character = "&"
+            let nestedKeyStartingCharacter: Character = "["
+            let nestedKeyEndingCharacter: Character = "]"
+            
+            func nestedKey(from deepObjectKey: String.SubSequence) -> Raw {
+                var unescapedFirstValue = Substring(deepObjectKey.removingPercentEncoding ?? "")
+                let nestedKey = unescapedFirstValue.parseBetweenCharacters(
+                    startingCharacter: nestedKeyStartingCharacter,
+                    endingCharacter: nestedKeyEndingCharacter
+                )
+                return nestedKey
+            }
+            
+            while !data.isEmpty {
+                let (firstResult, firstValue) = data.parseUpToEitherCharacterOrEnd(
+                    first: keyValueSeparator,
+                    second: pairSeparator
+                )
+                let key: Raw
+                let value: Raw
+                switch firstResult {
+                case .foundFirst:
+                    // Hit the key/value separator, so a value will follow.
+                    let secondValue = data.parseUpToCharacterOrEnd(pairSeparator)
+                    
+                    key = nestedKey(from: firstValue)
+                    value = secondValue
+                case .foundSecondOrEnd:
+                    // No key/value separator, treat the string as the key.
+                    key = nestedKey(from: firstValue)
+                    value = .init()
+                }
```

Comment:
Can you give an example for this case? The spec is light on examples and IIUC doesn't have an example that _isn't_ the `object[key]=value` form.

Reply:
Yes let's be strict here and throw an error if the string seems malformed.